src/include/execution/sql/updater.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+#include "execution/exec/execution_context.h"
+#include "execution/util/execution_common.h"
+namespace terrier::execution::sql {
+
+/**
+ * Helper class to perform updates in SQL Tables.
+ */
+class EXPORT Updater {
+ public:
+  /**
+   * Constructor
+   * @param exec_ctx The execution context.
+   * @param table_oid Oid of the table
+   * @param col_oids Col oids to updated.
+   * @param num_oids Number of column oids.
+   * @param is_index_key_update Whether this will update indexed keys.
+   */
+  explicit Updater(exec::ExecutionContext *exec_ctx, catalog::table_oid_t table_oid, uint32_t *col_oids,
+                   uint32_t num_oids, bool is_index_key_update);
+
+  /**
+   * Destructor.
+   */
+  ~Updater();
+
+  /**
+   * @return The table's projected row.
+   */
+  storage::ProjectedRow *GetTablePR();
+
+  /**
+   * @param index_oid oid of the index to access
+   * @return The projected row of the index.
+   */
+  storage::ProjectedRow *GetIndexPR(catalog::index_oid_t index_oid);
+
+  /**
+   * Update a tuple in the table.
+   * @param table_tuple_slot tuple slot of the tuple.
+   * @return Whether update was successful.
+   */
+  bool TableUpdate(storage::TupleSlot table_tuple_slot);
+
+  /**
+   * Delete a tuple from the table.
+   * @param table_tuple_slot tuple slot of the tuple.
+   * @return Whether delete was successful.
+   */
+  bool TableDelete(storage::TupleSlot table_tuple_slot);
```

Comment:
Does the `Updater` need `Delete` and `Insert` functions?  I think the binder and/or optimizer should handle rewriting an update on an indexed attribute to being a `select`, `delete`, `insert` sequence.  If that is the case, then these functions are unnecessary.